modules/RTCStats/RTCStats.ts

```diff
@@ -0,0 +1,189 @@
+
+const RTC_STATS_WC_DISCONNECTED = 'rtcstats_ws_disconnected';
+const RTC_STATS_PC_EVENT = 'rtstats_pc_event';
+
+/**
+ * Filter out RTCPeerConnection that are created by callstats.io.
+ *
+ * @param {*} config - Config object sent to the PC c'tor.
+ * @returns {boolean}
+ */
+function connectionFilter(config) {
+    if (config?.iceServers[0] && config.iceServers[0].urls) {
+        for (const iceUrl of config.iceServers[0].urls) {
+            if (iceUrl.includes('callstats.io')) {
+                return true;
+            }
+        }
+    }
+}
+
+/**
+ * RTCStats Singleton that is initialized only once for the lifetime of the app, subsequent calls to init will be ignored.
+ * Config and conference changes are handled by the start method.
+ */
+class RTCStats {
+    private _initialized: boolean = false;
+    private trace: any = null;
+    public events: EventEmitter = new EventEmitter();
+
+    /**
+     * RTCStats "proxies" WebRTC functions such as GUM and RTCPeerConnection by rewriting the global objects.
+     * The proxies will then send data to the rtcstats server via the trace object.
+     * The initialization procedure must be called once when lib-jitsi-meet is loaded.
+     * 
+     * @param initConfig initial config for rtcstats.
+     * @returns {void}
+     */
+    init(initConfig: any) {        
+        const {
+            analytics: {
+                rtcstatsUseLegacy: useLegacy = false,
+                rtcstatsPollInterval: pollInterval= 10000,
+                rtcstatsSendSdp: sendSdp = false,
+                rtcstatsEnabled = false
+            } = {}
+        } = initConfig;
+
+        // If rtcstats is not enabled or already initialized, do nothing.
+        // Calling rtcsatsInit multiple times will cause the global objects to be rewritten multiple times,
+        // with unforseseen consequences.
+        if (!rtcstatsEnabled || this._initialized) return;
+        
+        rtcstatsInit( 
+            { statsEntry: this.sendStatsEntry.bind(this) }, 
+            { connectionFilter,
+              pollInterval,
+              useLegacy,
+              sendSdp,
+              eventCallback: (event) => this.events.emit(RTC_STATS_PC_EVENT, event)}
+        );
+
+        this._initialized = true;
+    }
+
+    /**
+     * When a conference is about to start, we need to reset the trace module, and initialize it with the
+     * new conference's config. On a normal conference flow this wouldn't be necessary, as the whole page is
+     * reloaded, but in the case of breakout rooms or react native the js context doesn't reload, hence the
+     * RTCStats singleton and its config persists between conferneces.
+     * 
+     * @param conference - JitsiConference instance that's about to start.
+     * @returns {void}
+     */
+    start(conference: JitsiConference) {
+        // If rtcstats proxy module is not initialized, do nothing.
+        if (!this._initialized) return;
```

Comment:
When would this happen?

Reply:
Ideally it wouldn't, it's just meant as some basic invariant checking. But if you prefer a fail fast approach I can remove it.

Reply:
Let's add a log line at least, since it implies something is bad.

Reply:
Added.